### PR TITLE
[feature-wip](executor)query cpu hard limit based on doris scheduler

### PR DESCRIPTION
**Pull request by wangbo**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exec/scan/scanner_scheduler.cpp [hunk 1]
```diff
@@ -447,9 +446,19 @@ void ScannerScheduler::_task_group_scanner_scan(ScannerScheduler* scheduler,
...
+            if (!scan_task.is_empty_task) {
+                RuntimeProfile::Counter tmp_timer(TUnit::TIME_NS);
+                {
+                    SCOPED_CPU_TIMER(&tmp_timer);
+                    scan_task.scan_func();
+                }
+                time_spent = tmp_timer.value();
+            } else {
+                {
+                    SCOPED_RAW_TIMER(&time_spent);
```
Comment (HappenLee) [inline be/src/vec/exec/scan/scanner_scheduler.cpp:458]: better use a new timer to register the sleep timer for the debug target

File: be/src/vec/exec/scan/scan_task_queue.cpp [hunk 2]
```diff
@@ -132,6 +205,31 @@ void ScanTaskTaskGroupQueue::update_tg_cpu_share(const taskgroup::TaskGroupInfo&
...
+
+    // 0 <= user_g_cpu_hard_limit <= 100, bound by FE
+    // user_g_cpu_hard_limit = 0 means no group exists
+    int empty_group_cpu_share = 100 - user_g_cpu_hard_limit;
+    if (empty_group_cpu_share > 0 && empty_group_cpu_share < 100 && !contains_empty_group) {
+        _empty_group_entity->update_empty_cpu_share(empty_group_cpu_share);
+        _enqueue_task_group(_empty_group_entity);
+    } else if ((empty_group_cpu_share == 0 || empty_group_cpu_share == 100) &&
+               contains_empty_group) {
+        _dequeue_task_group(_empty_group_entity);
```
Comment (HappenLee) [inline be/src/vec/exec/scan/scan_task_queue.cpp:227]: the op is dangerous，do not set the empty cpu share， so `empty_group_entity` the `cpu_share` is undefine

File: be/src/vec/exec/scan/scan_task_queue.cpp [hunk 3]
```diff
@@ -78,9 +141,16 @@ bool ScanTaskTaskGroupQueue::take(ScanTask* scan_task) {
...
         }
     }
+    if (entity->is_empty_group_entity()) {
+        *scan_task = *_empty_scan_task;
+        return true;
+    }
     DCHECK(entity->task_size() > 0);
     if (entity->task_size() == 1) {
         _dequeue_task_group(entity);
+        if (_enable_cpu_hard_limit) {
```
Comment (HappenLee) [inline be/src/vec/exec/scan/scan_task_queue.cpp:151]: better impl the code in `dequeue` and `enqueue`

**Comment by wangbo** [comment]
run buildall

**Comment by wangbo** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]

**Comment by xiaokang** [comment]
2.0 wait for next pr

**Pull request merged**